src/main.cpp

```diff
@@ -49,23 +53,17 @@ constexpr unsigned long update_buzzer_controller_period_us = 100000; // 100 000
 constexpr unsigned long kick_watchdog_period_us = 10000;             // 10 000 us = 100 Hz
 constexpr unsigned long ams_update_period_us = 10000;                // 10 000 us = 100 Hz
 constexpr unsigned long ethernet_update_period = 10000;
-// from https://github.com/arkhipenko/TaskScheduler/wiki/API-Task#task note that we will use
-TsTask suspension_CAN_send(4000, TASK_FOREVER, &handle_enqueue_suspension_CAN_data, &task_scheduler,
-                           false);
-TsTask adc_0_sample_task(adc_sample_period_us, TASK_FOREVER, &run_read_adc0_task, &task_scheduler,
-                         false, &init_read_adc0_task);
-TsTask adc_1_sample_task(adc_sample_period_us, TASK_FOREVER, &run_read_adc1_task, &task_scheduler,
-                         false, &init_read_adc1_task);
-TsTask update_buzzer_controller_task(adc_sample_period_us, TASK_FOREVER,
-                                     &run_update_buzzer_controller_task, &task_scheduler, false);
-TsTask kick_watchdog_task(kick_watchdog_period_us, TASK_FOREVER, &run_kick_watchdog,
-                          &task_scheduler, false, &create_watchdog);
-TsTask ams_system_task(ams_update_period_us, TASK_FOREVER, &run_ams_system_task, &task_scheduler,
-                       false, &init_ams_system_task);
-
-TsTask CAN_send(TASK_IMMEDIATE, TASK_FOREVER, &handle_send_all_data, &task_scheduler, false);
-TsTask ethernet_send(ethernet_update_period, TASK_FOREVER, &handle_send_VCR_ethernet_data,
-                     &task_scheduler, false);
+constexpr unsigned long suspension_can_period_us = 4000;             // 4000 us = 250 Hz
+
+// task declarations
+HT_TASK::Task adc_0_sample_task(init_read_adc0_task, run_read_adc0_task, 1, adc_sample_period_us);
```

Comment:
I'd make the ADC0 priority 5, sampled at 4kHz, but I'd have ADC1 be priority 50 sampled at 100Hz. ADC1 doesn't have any important signals (doesn't have any signals rn, actually) so there's no reason to make it high priority with a fast update.